packages/node/test/integration/take-action.spec.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  Address,
+  AppABIEncodings,
+  AppState,
+  AssetType,
+  Node as NodeTypes
+} from "@counterfactual/types";
+import { Contract, ContractFactory, Wallet } from "ethers";
+import { AddressZero, One, Zero } from "ethers/constants";
+import { BaseProvider, Web3Provider } from "ethers/providers";
+import { bigNumberify, hexlify, randomBytes, SigningKey } from "ethers/utils";
+import FirebaseServer from "firebase-server";
+import ganache from "ganache-core";
+import { v4 as generateUUID } from "uuid";
+
+import TicTacToeApp from "../../../apps/build/TicTacToeApp.json";
+import {
+  IMessagingService,
+  InstallMessage,
+  IStoreService,
+  Node,
+  NODE_EVENTS,
+  NodeConfig,
+  ProposeMessage,
+  TakeActionMessage
+} from "../../src";
+import { ERRORS } from "../../src/methods/errors";
+
+import TestFirebaseServiceFactory from "./services/firebase-service";
+import {
+  EMPTY_NETWORK,
+  generateGetStateRequest,
+  generateTakeActionRequest,
+  getNewMultisig,
+  makeInstallRequest
+} from "./utils";
+
+describe("Node method follows spec - takeAction", () => {
+  let firebaseServiceFactory: TestFirebaseServiceFactory;
+  let firebaseServer: FirebaseServer;
+  let messagingService: IMessagingService;
+  let nodeA: Node;
+  let storeServiceA: IStoreService;
+  let nodeB: Node;
+  let storeServiceB: IStoreService;
+  let nodeConfig: NodeConfig;
+  let provider: BaseProvider;
+  let tttContract: Contract;
+  let privateKey: string;
+
+  beforeAll(async () => {
+    firebaseServiceFactory = new TestFirebaseServiceFactory(
+      process.env.FIREBASE_DEV_SERVER_HOST!,
+      process.env.FIREBASE_DEV_SERVER_PORT!
+    );
+    firebaseServer = firebaseServiceFactory.createServer();
+    messagingService = firebaseServiceFactory.createMessagingService(
+      process.env.FIREBASE_MESSAGING_SERVER_KEY!
+    );
+    nodeConfig = {
+      STORE_KEY_PREFIX: process.env.FIREBASE_STORE_PREFIX_KEY!
+    };
+
+    privateKey = new SigningKey(hexlify(randomBytes(32))).privateKey;
+    provider = new Web3Provider(
+      ganache.provider({
+        accounts: [
+          {
+            balance: "7fffffffffffffff",
+            secretKey: privateKey
+          }
+        ]
+      })
+    );
+
+    const wallet = new Wallet(privateKey, provider);
+    tttContract = await new ContractFactory(
+      TicTacToeApp.interface,
+      TicTacToeApp.bytecode,
+      wallet
+    ).deploy();
+  });
```

Comment:
You can abstract this whole `beforeAll` thing into some kind of `blockchain-tests-setup-helper` file, it is duplicated many times

Reply:
Yeah this duplication is undesirable. I'll refactor this.

Reply:
I'll clean this up along with https://github.com/counterfactual/monorepo/pull/548/files#r248885645